extensions/wikia/FilePage/FilePageHelper.class.php

```diff
@@ -67,10 +70,66 @@ public static function getFilePageRedirect( Title $title ) {
 				'file' => $title->getText()
 			] );
 		}
-		$wgMemc->set( $redirKey, $url );
+		if( !$onlyDB ) {
+			$wgMemc->set( $redirKey, $url );
+		}
 		return $url;
 	}
 
+	/**
+	 * Returns list of surrogate keys for purging.
+	 *
+	 * @requestParam Title
+	 *
+	 * @return array $keys - surrogate keys to purge
+	 */
+	public static function getSurrogateKeys( Title $title ) {
+		$keys = [];
+		$keys[] = self::getRedirSurrogateKey( $title );
+		if ( $title->inNamespace( NS_FILE ) ) {
+			$url = self::getFilePageRedirect( $title );
+			if( isset($url) ){
+				$pageTitle = self::rawURLToTitle( $url );
```

Comment:
as this converts any url to a local title object, I'd like to make sure - the url is always pointing to the current wiki, right?